src/transformStateWithClones.js

```diff
@@ -7,7 +7,24 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === `addProperties`) {
+      currentState = { ...currentState, ...action.extraData };
+    } else if (action.type === `removeProperties`) {
+      action.keysToRemove.forEach((key) => {
+        delete currentState[key];
+      });
+    } else if (action.type === `clear`) {
+      currentState = {};
+    }
```

Comment:
Consider using a `switch` statement instead of multiple `if/else if` statements when you have a limited amount of conditions to check for different `action.type` values. This will improve the readability and maintainability of the code. Remember to include a `default` case for error handling as per the checklist.